src/functions/Coverage.ps1

```diff
@@ -211,54 +211,35 @@ function Resolve-CoverageInfo {
 
     $filePaths = Get-CodeCoverageFilePaths -Paths $resolvedPaths -IncludeTests $includeTests -RecursePaths $recursePaths
 
-    $params = @{
+    $commonParams = @{
         StartLine = $UnresolvedCoverageInfo.StartLine
         EndLine   = $UnresolvedCoverageInfo.EndLine
         Class     = $UnresolvedCoverageInfo.Class
         Function  = $UnresolvedCoverageInfo.Function
     }
 
     foreach ($filePath in $filePaths) {
-        $params['Path'] = $filePath
-        New-CoverageInfo @params
+        New-CoverageInfo @commonParams -Path $filePath
     }
 }
 
 function Get-CodeCoverageFilePaths {
     param (
-        [object]$Paths,
+        [string[]]$Paths,
         [bool]$IncludeTests,
         [bool]$RecursePaths
     )
 
     $testsPattern = "*$($PesterPreference.Run.TestExtension.Value)"
 
-    $filePaths = foreach ($path in $Paths) {
-        $item = & $SafeCommands['Get-Item'] -LiteralPath $path
-        if ($item -is [System.IO.FileInfo] -and ('.ps1', '.psm1') -contains $item.Extension -and ($IncludeTests -or $item.Name -notlike $testsPattern)) {
-            $item.FullName
-        }
-        elseif ($item -is [System.IO.DirectoryInfo]) {
-            $children = foreach ($i in & $SafeCommands['Get-ChildItem'] -LiteralPath $item) {
-                # if we recurse paths return both directories and files so they can be resolved in the
-                # recursive call to Get-CodeCoverageFilePaths, otherwise return just files
-                if ($RecursePaths) {
-                    $i.PSPath
-                }
-                elseif (-not $i.PSIsContainer) {
-                    $i.PSPath
-                }
-            }
-            Get-CodeCoverageFilePaths -Paths $children -IncludeTests $IncludeTests -RecursePaths $RecursePaths
-        }
-        elseif (-not $item.PsIsContainer) {
-            # todo: enable this warning for non wildcarded paths? otherwise it prints a ton of warnings for documentation and so on when using "folder/*" wildcard
-            # & $SafeCommands['Write-Warning'] "CodeCoverage path '$path' resolved to a non-PowerShell file '$($item.FullName)'; this path will not be part of the coverage report."
+    [string[]] $filteredFiles = @(foreach ($file in (& $SafeCommands['Get-ChildItem'] -LiteralPath $Paths -File -Recurse:$RecursePaths)) {
+        if (('.ps1', '.psm1') -contains $file.Extension -and ($IncludeTests -or $file.Name -notlike $testsPattern)) {
+            $file.FullName
         }
-    }
-
-    return $filePaths
+    })
```

Comment:
Include or revert refactor? AFAIK it's the same behavior but faster and cleaner.
If yes, ignore in v5 backport to play safe?

Reply:
Let's avoid if you think it might not be safe.

Reply:
I'm fine with including in backport unless we find an issue. Free perf and easy revert in a patch release + same logic so it's easier to support both. I just thought you'd might want to avoid too many refactors in v5 so leaving decision to you 🙂

Reply:
I've kicked off the backport, at least it will unblock OP. And I will have something to release in v5 to try the releasing from server (once I port it back :"D)